src/fsharp/TypeChecker.fs

```diff
@@ -6821,8 +6821,8 @@ and TcRecdExpr cenv overallTy env tpenv (inherits, optOrigExpr, flds, mWholeExpr
             None,tpenv
 
     let expr,tpenv = 
-        let fldsList = fldsList |> List.choose (fun (n, v) -> if v.IsSome then Some (n, v.Value) else None)
-        TcRecordConstruction cenv overallTy env tpenv optOrigExpr  overallTy fldsList mWholeExpr
+        let fldsList = fldsList |> List.choose (fun (n, v) -> match v with Some v -> Some (n, v) | _ -> None)
```

Comment:
Nah, too much effort :D I'm just stalking github on my coffee break.
And I think I like the version by @vasily-kirichenko better than my own anyway, List Expressions are great.